### PR TITLE
ENG-404 - Update Codeline Component

### PR DESCRIPTION
**Pull request by dogmar**
[Linear Ticket](https://linear.app/pluralsh/issue/ENG-404/update-codeline-component)

Also updated the Tooltip component to use with Codeline

image

### CODE REVIEW

File: src/components/Codeline.tsx [hunk 1]
```diff
@@ -10,71 +14,85 @@ const propTypes = {}
...
       >
-        <CopyIcon />
+        <Tooltip
+          offset={8}
+          label="Copied!"
+          color="text-success-light"
+          placement="top"
+          displayOn="manual"
+          dismissable
+          onOpenChange={open => {
```
Comment (michaeljguarino) [inline src/components/Codeline.tsx:74]: we can technically one-line this right or do the linters complain?

File: src/components/Tooltip.tsx [hunk 2]
```diff
@@ -1,27 +1,235 @@
...
+  // Preserve the consumer's ref
+  const childrenRef = useMemo(() => mergeRefs([reference, (children as any).ref]),
+    [reference, children])
+
+  const arrowRotation = {
+    top: 0,
+    right: 90,
+    bottom: 180,
+    left: -90,
+  }[finalPlacement.split('-')[0]]
```
Comment (michaeljguarino) [inline src/components/Tooltip.tsx:159]: could you save this to a const so the split isn't double executed?

File: src/components/Codeline.tsx (hunk 1 above)
Comment (dogmar) [reply src/components/Codeline.tsx:74]: Linter complains.

**Review by maciaszczykm — APPROVED** [review]

File: src/components/Tooltip.tsx (hunk 2 above)
Comment (dogmar) [reply src/components/Tooltip.tsx]: Done.

**Pull request merged**